sdk/appconfiguration/app-configuration/src/appConfigurationClient.ts

```diff
@@ -128,18 +128,25 @@ export class AppConfigurationClient {
     key: string,
     options: GetConfigurationSettingOptions = {}
   ): Promise<GetConfigurationSettingResponse> {
-    const response = await this.client.getKeyValue(key, {
+    const response = (await this.client.getKeyValue(key, {
       label: options.label,
       ...options,
       ...checkAndFormatIfAndIfNoneMatch(options)
-    }) as GetConfigurationSettingResponse;
+    })) as GetConfigurationSettingResponse;
 
     // 304 only comes back if the user has passed a conditional option in their
     // request _and_ the remote object has the same etag as what the user passed.
     if (response._response.status === 304) {
-      throw new ResponseBodyNotFoundError("Remote resource matches local resource, no body returned.", "Resource same as remote", response._response.status, response._response.request, response._response, null);
+      throw new ResponseBodyNotFoundError(
```

Comment:
"ResponseBodyNotFound" seems like a misleading name for what's actually happening here, that the remote content has not changed since the last ETag the client received.  Why do we throw an error in this case anyway?

Reply:
Normally we have two paths after a request - the request succeeded and the body now contains the model information or we failed (in which case the body might contain anything but there's no expectation).

AppConfig adds a third one - the request "succeeded" but the body is not usable. So the choices of what do here were less clear. 

What we ended up with is an error type that can be used to distinguish that you're in this state (which you opt-in to by passing some specific parameters). So you still get an exception so you don't accidentally go and assume you can use some of the model's fields. 

In the future if we get 204/304 responses we have a way of indicating that the returned model isn't usable if that's what is happening (ie, body not found).

This is also just being consistent with what the other SDKs are now doing for appconfig.

CC: @bterlson

Reply:
At the very least I think the message you pass to the error constructor could be clearer, like "The requested setting's value has not changed since the last request."  That way the user who doesn't know they should catch an exception in this case would have some chance of understanding why it happened.

Reply:
I like that message, I'll change it.

Reply:
Also, quick note - the user opts-in to this behavior so hopefully the combination of that and the improved error message is enough to steer them where they need to go.